General rework

jensewe opened the pull request:
- [x] Remove unnecessary detours
- [x] Rework hook vectors to hashmaps
- [x] Multiplayer testing

jensewe committed:
Rework detours

jensewe committed:
Fix snapshot leaking memory

jensewe committed:
Enable optimization

jensewe committed:
Attempt to fix missing `ThreadInterlockedDecrement`

jensewe committed:
Attempt to fix missing `ThreadInterlockedDecrement`

jensewe committed:
Attempt to fix wrong player data

jensewe committed:
oops

jensewe committed:
oops2

jensewe committed:
Fix mistake clearing packed entities

jensewe committed:
Fix mistake clearing packed entities 2

jensewe committed:
Fix mistake clearing packed entities 3

jensewe committed:
Update extension.cpp

jensewe committed:
Fix mistake clearing packed entities 4

jensewe committed:
Fix mistake clearing packed entities 5

jensewe committed:
Revert all packed entity changes

jensewe committed:
Remove outdated binaries

jensewe committed:
Remove leftover line

jensewe committed:
Update AMBuildScript

SomeDeadGirl commented:
Will there be a complied DLL to download anytime soon? I would like to use a sourcemod plugin but it requires a sendproxy and I only use Windows and not Linux. Thanks.

jensewe committed:
Update windows gamedata & Cleanup

jensewe committed:
Remove leftover `CGameClient::s_iOffs_edict`

jensewe commented:
> Will there be a complied DLL to download anytime soon? I would like to use a sourcemod plugin but it requires a sendproxy and I only use Windows and not Linux. Thanks.

Just updated with full windows gamedata, it might be working now on windows.
I've semi-abanconed this project because there seems to be unconfirmed issues with entity origins (I don't really think it's related and nor could I find anything suspicious). Good luck if you decide to take it.

SomeDeadGirl commented:
> > Will there be a complied DLL to download anytime soon? I would like to use a sourcemod plugin but it requires a sendproxy and I only use Windows and not Linux. Thanks.
> 
> Just updated with full windows gamedata, it might be working now on windows. I've semi-abanconed this project because there seems to be unconfirmed issues with entity origins (I don't really think it's related and nor could I find anything suspicious). Good luck if you decide to take it.

I understand. Will I need to compile the extension folder to make the DLL file? Every time I try to do it in the Visual Studio, It comes across an error. I'm not knowledgeable in this field so.

jensewe commented:
Check the actions here: https://github.com/jensewe/Left4SendProxy/actions

SomeDeadGirl commented:
> Check the actions here: https://github.com/jensewe/Left4SendProxy/actions

So far, It's working flawlessly. The plugin that needed this dependency is functioning. I will need to wait for my friends to wake up to test in-case it causes a hidden error as that happened with SuperLoft/BTFighter Sendproxy Win Repo.
(The error would break survivors animations and later crash in an random interval on his sendproxy).

I will post sometime later tonight on the results. And if it doesn't work well, I understand you not wanting to continue any further. Thank you for your time.

SomeDeadGirl commented:
Yeah, It doesn't work with other players similar to SuperLoft/BTFighter, it will work fine with solo and with bots. The moment you try to play with other players, it will crash the game.

Unfortunate.... :(

jensewe commented:
> Yeah, It doesn't work with other players similar to SuperLoft/BTFighter, it will work fine with solo and with bots. The moment you try to play with other players, it will crash the game.
> 
> Unfortunate.... :(

Could you share the crash dump if accelerator is installed? Feel free to fix it atm.

SomeDeadGirl commented:
> > Yeah, It doesn't work with other players similar to SuperLoft/BTFighter, it will work fine with solo and with bots. The moment you try to play with other players, it will crash the game.
> > Unfortunate.... :(
> 
> Could you share the crash dump if accelerator is installed? Feel free to fix it atm.

I tried to log it but accelerator didn't log it whatsoever. Sorry. I was using this plugin. 
https://forums.alliedmods.net/showthread.php?p=2820685

If you can just grab 1 person with you on any map, for example Dead Center and it crashes. But if you play alone with bots it works fine.

jensewe commented:
Sorry for the late reply.
The plugin is unlikely to be the matter when the crash happens only when 2 or more players are present.

Unfortunately I don't have a setup for Windows testing in near future, appreciate if you could try again with Accelerator installed.
If somehow failing to upload the dump, feel free to attach the raw file in `addons/sourcemod/data/dumps` before the Accelerator is active again.

SomeDeadGirl commented:
> Sorry for the late reply. The plugin is unlikely to be the matter when the crash happens only when 2 or more players are present.
> 
> Unfortunately I don't have a setup for Windows testing in near future, appreciate if you could try again with Accelerator installed. If somehow failing to upload the dump, feel free to attach the raw file in `addons/sourcemod/data/dumps` before the Accelerator is active again.

[server-id.txt](https://github.com/user-attachments/files/17196786/server-id.txt)
[accelerator.log](https://github.com/user-attachments/files/17196752/accelerator.log)
here are the raw and accelerator log, thats all i got. i even removed my own plugins to make sure it wasnt conflicting. but it still made me crash. i dont think it says why it crashes though. i apologize for being late. it only makes me crash with other players not solo so i had to wait for my friends to be available to do it.

jensewe committed:
Fix unable to `UnhookArrayPropGamerules`

Thanks to @IA-NanaNana for reporting.

jensewe committed:
Update .gitignore

jensewe committed:
Attempt to address entity origin issue

jensewe committed:
Attempt to address entity origin issue

jensewe committed:
Merge branch 'main' into rework

jensewe committed:
Merge branch 'rework' of https://github.com/jensewe/Left4SendProxy into rework

jensewe committed:
Reworks

jensewe committed:
Merge branch 'main' into rework

jensewe committed:
Fix missing "sm_hashmap.h"

jensewe committed:
Fix missing hash policy

jensewe committed:
Cleanup

jensewe committed:
Fix SM1.11 compatibility

jensewe committed:
Fix callback type of gamerules hooks

jensewe committed:
Fix assertion failure on current client index

jensewe committed:
Enable debug build

jensewe committed:
Fix wrong element passed to callback when hooking datatable member

NullExir commented:
Hi, can you please enable issues?, I would like to report a bug with this extension

jensewe commented:
> Hi, can you please enable issues?, I would like to report a bug with this extension

Done. Make sure to specify the version (main/rework) that the issue is on.

jensewe committed:
Fix some mistakes causing crashes

Address #3

jensewe committed:
Fix startup crash on windows

jensewe committed:
Remove currently unnecessary interfaces

jensewe committed:
Deprecate use on local server

jensewe committed:
Improve documentation in include file

jensewe committed:
Restore `SMEXT_ENABLE_FORWARDSYS`

jensewe committed:
Remove client assertion to prevent crashes

Prevent crashes due to client assertion failure if props get hooked before "OnMapStart".

jensewe committed:
Fix client 0 in string proxy callback

jensewe committed:
Bump version & Update info

jensewe committed:
Revert fatal assertion in `HookEntity`

jensewe committed:
Fix frame snapshot leaking

jensewe committed:
Fix obsolete packs cause crashes on new client joining

Fix:
- Dereference entity packs of leaving clients, so later joining clients that reuse slots won't crash the server for mismatching prop count during entity packing.

Change:
- Save edict change state for each client instead of forcing repack whenever there's a newer pack, which causes enormous entity pack count in total (and thus leads to "CUtlRBTree overflow" crashes).
- Prefer `at` over `operator[]` on STL map to probably reduce some overhead.

jensewe committed:
Fix crash on unloading

jensewe committed:
Minor optimization on save/apply change state

jensewe committed:
Allow returning invalid ehandle

jensewe committed:
Minor code reordering

jensewe committed:
Cleanup

jensewe committed:
Minor include reorganizing

jensewe committed:
Enable Valve's memory override

jensewe committed:
Update Python version to 3.12

jensewe committed:
Revert "Enable Valve's memory override"

This reverts commit eca27e229ba019e764f2a507adcfd5fe0e1fb13f.

jensewe committed:
Attempt to fix invalid memory access on windows

jensewe committed:
Revert "Attempt to fix invalid memory access on windows"

This reverts commit 0d4368d06604f4f515e1dba2f4828028a155bac6.

jensewe committed:
Enable Valve's memory override

jensewe committed:
Disable debug build for Windows

due to unhandled error LNK2005 with VS2022.

jensewe committed:
Reconcile folder structure

jensewe committed:
Fix source codes being included to package

jensewe committed:
Update AMBuild to 2.2

jensewe committed:
Attempt to add safetyhook support

jensewe committed:
Add build for multiple SM versions

jensewe committed:
Merge branch 'master' into rework

jensewe committed:
Fix build step & name

jensewe committed:
Merge branch 'rework' of https://github.com/jensewe/Left4SendProxy into rework

jensewe committed:
Attempt to fix hl2sdk manifest lookup

jensewe committed:
Fix build

jensewe committed:
Fix build with SM1.11

jensewe committed:
Fix build artifact name

jensewe committed:
Restore local server support except when network backdoor is used

jensewe committed:
workflow: Remove L4D1 build to prevent confusion

At the moment gamedata for L4D1 is not ready yet.

jensewe committed:
Fix compile errors

jensewe committed:
Notify user when used in single player

jensewe committed:
.gitignore: Add safetyhook folder
